LoopTests/Mock Stores/MockDoseStore.swift

```diff
@@ -0,0 +1,159 @@
+        self.insulinDeliveryStore = InsulinDeliveryStore(
+            healthStore: HKHealthStoreMock(),
+            observeHealthKitForCurrentAppOnly: false,
+            cacheStore: PersistenceController(directoryURL: URL.init(fileURLWithPath: "")),
+            observationEnabled: true,
+            test_currentDate: MockDoseStore.currentDate(for: test)
+        )
+        self.pumpEventQueryAfterDate = MockDoseStore.currentDate(for: test)
+        self.lastAddedPumpData = MockDoseStore.currentDate(for: test)
+    }
+    
+    static let dateFormatter = ISO8601DateFormatter.localTimeDate()
+    
+    var testType: DataManagerTestType
+    
+    var basalProfileApplyingOverrideHistory: BasalRateSchedule?
+    
+    var insulinDeliveryStore: InsulinDeliveryStore
+    
+    var delegate: DoseStoreDelegate?
+    
+    var device: HKDevice?
+    
+    var pumpRecordsBasalProfileStartEvents: Bool = false
+    
+    var pumpEventQueryAfterDate: Date
+    
+    var basalProfile: BasalRateSchedule?
+    
+    // Default to the adult exponential insulin model
+    var insulinModel: InsulinModel? = ExponentialInsulinModelPreset.humalogNovologAdult
+    
+    var insulinSensitivitySchedule: InsulinSensitivitySchedule?
+    
+    var sampleType: HKSampleType?
+    
+    var authorizationRequired: Bool = false
+    
+    var sharingDenied: Bool = false
+    
+    var lastReservoirValue: ReservoirValue?
+    
+    var lastAddedPumpData: Date
+    
+    func addPumpEvents(_ events: [NewPumpEvent], lastReconciliation: Date?, completion: @escaping (DoseStore.DoseStoreError?) -> Void) {
+        completion(nil)
+    }
+    
+    func addReservoirValue(_ unitVolume: Double, at date: Date, completion: @escaping (ReservoirValue?, ReservoirValue?, Bool, DoseStore.DoseStoreError?) -> Void) {
+        completion(nil, nil, false, nil)
+    }
+    
+    func insulinOnBoard(at date: Date, completion: @escaping (DoseStoreResult<InsulinValue>) -> Void) {
+        completion(.failure(.configurationError))
+    }
+    
+    func generateDiagnosticReport(_ completion: @escaping (String) -> Void) {
+        completion("")
+    }
+    
+    func resetPumpData(completion: ((DoseStore.DoseStoreError?) -> Void)?) {
+        completion?(.configurationError)
+    }
+    
+    func getInsulinOnBoardValues(start: Date, end: Date?, basalDosingEnd: Date?, completion: @escaping (DoseStoreResult<[InsulinValue]>) -> Void) {
+        completion(.failure(.configurationError))
+    }
+    
+    func getNormalizedDoseEntries(start: Date, end: Date?, completion: @escaping (DoseStoreResult<[DoseEntry]>) -> Void) {
+        completion(.failure(.configurationError))
+    }
+    
+    func executePumpEventQuery(fromQueryAnchor queryAnchor: DoseStore.QueryAnchor?, limit: Int, completion: @escaping (DoseStore.PumpEventQueryResult) -> Void) {
+        completion(.failure(DoseStore.DoseStoreError.configurationError))
```

Comment:
I didn't have these functions to return any meaningful values because they aren't used in the portion of the algo that I'm testing (but they could be updated later if needed for other LDM tests)